doc/nrf/releases/release-notes-2.2.0.rst

```diff
@@ -0,0 +1,982 @@
+    This allows cancelling any update packages even if they are already marked to be updated.
+
+MPSL libraries
+--------------
+
+* Added :ref:`mpsl_lib`.
+
+sdk-nrfxlib
+-----------
+
+See the changelog for each library in the :doc:`nrfxlib documentation <nrfxlib:README>` for additional information.
+
+Scripts
+=======
+
+* :ref:`west_sbom`:
+
+  * Updated the SPDX License List database to version 3.18.
+
+* :ref:`partition_manager`:
+
+  * Added:
+
+    * :kconfig:option:`CONFIG_PM_PARTITION_ALIGN_SETTINGS_STORAGE` Kconfig option to specify the alignment of the settings storage partition.
+    * :kconfig:option:`CONFIG_PM_EXTERNAL_FLASH_HAS_DRIVER` Kconfig option.
+      This option must be selected by drivers providing support for external flash.
+      It is automatically selected by :kconfig:option:`CONFIG_NORDIC_QSPI_NOR` and :kconfig:option:`CONFIG_SPI_NOR`.
+      MCUboot might fail to boot when the external flash memory is used for non-primary application images and when the driver for the external flash memory is not enabled.
+      See :ref:`ug_bootloader_external_flash` and :ref:`pm_external_flash` for details.
+    * P-GPS partition section to the :file:`ncs/nrf/subsys/partition_manager/Kconfig` file.
+    * Board-specific static Partition Manager configuration shared among build types.
+      The configuration can be defined in the board's directory.
+
+  * Updated the :file:`ncs/nrf/subsys/partition_manager/pm.yml.pgps` file to place P-GPS partition in external flash when so configured.
+
+  * Fixed an issue with the driver and devicetree symbol for the external flash memory where the driver was sometimes NULL, even if the DT node was chosen.
+
+    .. note::
+       If your project includes MCUboot and uses external flash for storing secondary images, after applying this fix, you need to configure ``nordic,pm-ext-flash`` property in a devicetree overlay for the MCUboot child image.
+       Otherwise, MCUboot will not be able to read a flash area in the external flash to pick up a new firmware image.
+
+* Unity:
+
+  * Added:
+
+    * Support for excluding functions from mock generation.
+      This provides a framework to implement custom mocks.
+
+  * Updated:
+
+    * The generated mock functions have been renamed from ``__wrap`` to ``__cmock``.
+    * The generated mock headers have been renamed from :file:`mock_<header_file>.h` to :file:`cmock_<header_file>.h`.
+    * The compiler option ``--defsym`` is used instead of ``--wrap``.
+
+MCUboot
+=======
+
+The MCUboot fork in |NCS| (``sdk-mcuboot``) contains all commits from the upstream MCUboot repository up to and including ``cfec947e0f8be686d02c73104a3b1ad0b5dcf1e6``, with some |NCS| specific additions.
+
+The code for integrating MCUboot into |NCS| is located in the :file:`ncs/nrf/modules/mcuboot` folder.
+
+The following list summarizes both the main changes inherited from upstream MCUboot and the main changes applied to the |NCS| specific additions:
+
+* Added:
+
```

Comment:
* added precise check of the image size

Reply:
Added

Reply:
- loader: Added post copy hook to swap function
- fixed ram loading for ARM, with correct handling of vector table when code has moved to RAM

Reply:
Added these entries. please check.

Reply:
@nordicjm You probably want to mention https://github.com/nrfconnect/sdk-mcuboot/commit/11c5149a7dd6c9f7581e971656bae461c135e31a here

Reply:
Already covered under DFU and FOTA

Reply:
Serial recovery: Fixed issue with QSPI stack alignment which prevented writing data to QSPI slots.
Serial recovery: Fixed wrongly returning rc as an unsigned integer
Serial recovery: Added configuration option to configure size of previously 512-byte fixed buffer, allowing to receive larger packets which speed up data transfer

Reply:
Added